lib/hermetic/youtrack.ex

```diff
@@ -59,32 +59,40 @@ defmodule Hermetic.YouTrack do
     base_url() <> "/static/apple-touch-icon-180x180.png"
   end
 
-  @doc ~S"""
-  Fetch list of all available YouTrack project IDs.
-  """
-  @spec project_ids :: list(String.t())
-  def project_ids do
-    json = Jason.decode!(request(:get, "/rest/project/all").body)
-    for %{"shortName" => id} <- json, do: id
+  defmodule ProjectID do
+    use LambdaCache, name: __MODULE__
+
+    @doc ~S"""
+    Fetch list of all available YouTrack project IDs.
+    """
+    @spec refresh :: list(String.t())
+    def refresh do
+      json = Jason.decode!(YouTrack.request(:get, "/rest/project/all").body)
```

Comment:
Not good. Let's patch HTTPoison if you want whole-response transformations. Otherwise it will be messy forever.

Reply:
Also variable name is not descriptive: it says `json` when it stores already decoded term (`json` would be a fine name if it stored a JSON string). Should be `data`, or `payload`, or something along those lines.

Reply:
I'll patch HTTPoison another time in another PR. I made an issue in YouTrack for it. (OPS-504)